package.json

```diff
@@ -13,5 +13,10 @@
 	"devDependencies": {
 		"prettier": "^3.3.3",
 		"syncpack": "^13.0.0"
+	},
+	"pnpm": {
+		"overrides": {
+			"vite": "/Users/dario/Repos/my-repos/vite/packages/vite"
```

Comment:
Naturally this only works locally, I will push my stuff on my vite fork in a bit 🙂

Reply:
It's great this requires so few changes to Vite! Is the idea that this is something you could opt in to in the environment config somehow (and we could set this option for the environments created by our plugin)?

Reply:
> Is the idea that this is something you could opt in to in the environment config somehow (and we could set this option for the environments created by our plugin)?

Yes and yes, it's practically opt-in and we could set this in our plugin (this could still be overridden by the plugin's consumer)

To opt-in you simply need to let vite know where it should start its pre-bundling discovering crawling, and that's actually what I added here: https://github.com/flarelabs-net/vite-plugin-cloudflare/pull/32/files#diff-ccdf400ac3f84526b65f252b32659a7a52e1bfbc70d008e1baf4f9a98c87b9f3R129-R134

Reply:
Ah, OK. I wasn't sure if interpreting the existing `optimizeDeps.entries` option in this way would be a breaking change. What happens in current Vite if you provide this option for a custom environment?

Reply:
The above is no longer applicable since the way the ssr deps crawling is triggered has changed, now to enable it you need to set `noDiscovery: false` alongside the `optimizeDeps.entries` or `rollupOptions.input` values

(also `resolve.noExternal: true` to actually make sure that the dependencies do get pre-bundled)

So there is no breaking change concern 🙂